File: maratona-sbc-2016/2233/solve.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
	long long int r,g,b;
	cin >> hex >> r >> g >> b;
	cout << hex << 1 + (g > r ? 0 : ( (r/g) * (r/g) ) ) + ((g > r ? 0 : ( (r/g) * (r/g) ) ) * (g/b) * (g/b)) << endl;
	return 0;
}

File: maratona-sbc-2016/2233/solve.py
r = int(input(), 16)
g = int(input(), 16)
b = int(input(), 16)
G = (r//g) ** 2
if g > r:
    G = 0
B = G * ((g//b) ** 2)
print(hex(1 + G + B).split('x')[-1])